sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -240,63 +233,62 @@ static Read create() {
+     * determine the instance.
+     *
+     * <p>Does not modify this object.
+     */
+    public Read withProjectId(String projectId) {
+      return withProjectId(ValueProvider.StaticValueProvider.of(projectId));
+    }
+
     /**
      * Returns a new {@link BigtableIO.Read} that will read from the Cloud Bigtable instance
      * indicated by given parameter, requires {@link #withProjectId(String)} to be called to
      * determine the project.
      *
      * <p>Does not modify this object.
      */
-    public Read withInstanceId(String instanceId) {
+    public Read withInstanceId(ValueProvider<String> instanceId) {
       BigtableConfig config = getBigtableConfig();
       return toBuilder().setBigtableConfig(config.withInstanceId(instanceId)).build();
     }
 
     /**
-     * WARNING: Should be used only to specify additional parameters for connection
-     * to the Cloud Bigtable, instanceId and projectId should be provided over
-     * {@link #withInstanceId(String)} and {@link #withProjectId(String)} respectively.
-     *
-     * <p>Returns a new {@link BigtableIO.Read} that will read from the Cloud Bigtable instance
-     * indicated by {@link #withProjectId(String)}, and using any other specified customizations.
+     * Returns a new {@link BigtableIO.Read} that will read from the Cloud Bigtable instance
+     * indicated by given parameter, requires {@link #withProjectId(String)} to be called to
+     * determine the project.
      *
      * <p>Does not modify this object.
-     *
-     * @deprecated will be replaced by bigtable options configurator.
      */
-    @Deprecated
-    public Read withBigtableOptions(BigtableOptions options) {
-      checkArgument(options != null, "options can not be null");
-      return withBigtableOptions(options.toBuilder());
+    public Read withInstanceId(String instanceId) {
+      return withInstanceId(ValueProvider.StaticValueProvider.of(instanceId));
     }
 
     /**
-     * WARNING: Should be used only to specify additional parameters for connection to
-     * the Cloud Bigtable, instanceId and projectId should be provided over
-     * {@link #withInstanceId(String)} and {@link #withProjectId(String)} respectively.
-     *
-     * <p>Returns a new {@link BigtableIO.Read} that will read from the Cloud Bigtable instance
-     * indicated by the given options, and using any other specified customizations.
-     *
-     * <p>Clones the given {@link BigtableOptions} builder so that any further changes
-     * will have no effect on the returned {@link BigtableIO.Read}.
+     * Returns a new {@link BigtableIO.Read} that will read from the specified table.
      *
      * <p>Does not modify this object.
-     *
-     * @deprecated will be replaced by bigtable options configurator.
      */
-    @Deprecated
-    public Read withBigtableOptions(BigtableOptions.Builder optionsBuilder) {
+    public Read withTableId(ValueProvider<String> tableId) {
```

Comment:
Unfortunately, we need to keep public versions of `withBigtableOptions()` and `getBigtableOptions()`.  We need to maintain backwards compatible APIs